jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebAppClassLoader.java

```diff
@@ -543,27 +543,17 @@ else if (e != ex)
             else
             {
                 // Not parent loader priority, so...
-
-                // Try the webapp classloader first
-                // Look in the webapp classloader as a resource, to avoid 
-                // loading a system class.
-                String path = name.replace('.', '/').concat(".class");
-                URL webapp_url = findResource(path);
-                
-                if (webapp_url!=null && !_context.isSystemResource(name,webapp_url))
+                webapp_class = loadAsResource(name, true);
+                if (webapp_class != null)
```

Comment:
Don't we need to do a `resolveClass(webapp_class);` before returning it below?

Reply:
The resolveClass() is being done in loadAsResource() method, which preserves exactly the circumstances in which resolveClass() was called before, nothing has changed in that logic.